lib/vector-core/proto/event.proto

```diff
@@ -93,6 +93,7 @@ message Metadata {
   optional string source_type = 4;
   OutputId upstream_id = 5;
   Secrets secrets = 6;
+  bytes internal_event_id = 7;
```

Comment:
Nit: we could call `source_trace_id`

Reply:
I'd say `source_event_id`, but agree on not `internal_`